pkg/executor/wasm/handler.go

```diff
@@ -83,33 +80,20 @@ func (h *executionHandler) run(ctx context.Context) {
 		h.cancel()
 	}()
 
-	configuredProtocol := strings.ToLower(os.Getenv("OTEL_EXPORTER_OTLP_PROTOCOL"))
-	endpoint := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
-
-	protocol := opentelemetry.GRPC
-	if configuredProtocol != "grpc" {
-		protocol = opentelemetry.HTTP
-	}
-
-	allowInsecure := false
-	if u, err := url.Parse(endpoint); err == nil {
-		host := u.Hostname()
-		endpoint = host + ":" + u.Port()
-		allowInsecure = host == "localhost" || host == "127.0.0.1"
-	}
-
-	conf := &opentelemetry.OTelConfig{
+	var adapter *opentelemetry.OTelAdapter
+	conf := opentelemetry.OTelConfig{
 		ServiceName:        "bacalhau",
 		EmitTracesInterval: time.Second * 1,
-		TraceBatchMax:      100,
-		Endpoint:           endpoint,
-		Protocol:           protocol,
-		AllowInsecure:      allowInsecure, // for localhost in dev via http
+		TraceBatchMax:      10,
+		// the remaining fields are completed from a system-configured client
+		// by using the `UseCustomClient` method on the adapter below
+	}
+	traceClient, err := telemetry.GetTraceClient()
```

Comment:
@simonwo - we've made this function visible in order to create another instance of the otel client (which is more consistently configured from the same sources). Initially, we tried to set the existing client to a package-level global and re-use a single instance of it, but it seems that there is some context cancellation on a per-trace basis which caused errors between the wasm spans and the root spans.

Reply:
Although maybe it would be better to put as much of our adapter initialization code in the `telemetry` package and construct adapters there to be used in the `tracedRuntime` struct.